CRM/Mailing/BAO/TrackableURL.php

```diff
@@ -54,10 +54,11 @@ private static function getBasicTrackerURL($url, $mailing_id, $queue_id) {
     }
 
     // hack for basic CRM-1014 and CRM-1151 and CRM-3492 compliance:
-    // let's not replace possible image URLs and CiviMail ones
+    // let's not replace possible image URLs, CiviMail URLs or internal anchor URLs
     if (preg_match('/\.(png|jpg|jpeg|gif|css)[\'"]?$/i', $url)
       or substr_count($url, 'civicrm/extern/')
       or substr_count($url, 'civicrm/mailing/')
+      or ($url[0] === '#')
```

Comment:
@larssandergreen  would be worth it to add here another condition as ?:
`or substr_count($url, 'mailto:')`

Reply:
Is there a good reason not to track mailto: links? I'm in favour of keeping them in the clickthrough report, both in terms of counting them as a click for the overall rate and in the individual link statistics.
I guess for people who use a non-browser based client, there's a roundtrip to the browser and back for the clicked link, but is that a big problem?

Reply:
hmm ok, we don't see the value of that tracking those, specially when sending and email through a SMTP provider that tracks its own stats, so it's a triple jump to get at the end to compose an email. But seems fair to follow the discussion in a different issue/PR, than this one focused on local anchors

Reply:
FWIW, it looks like there was a previous endeavor track `mailto:`s when use BAO delivery -- eg @JKingsnorth's #7950. 

I guess when considering whether to track a URL, there will be a few buckets like:

* Cases where everyone agrees tracking is appropriate. (Outbound `http(s)` URLs)
* Cases where everyone agrees tracking is not appropriate. (Page-internal anchors)
* Cases where people may reasonably have different opinions (eg `mailto:`).

I'd be a soft +1 for some kind of setting or hook to address the latter.

Reply:
+1 to @totten 's +1 to a setting or hook to address mailto: (or even whatsapp: etc.) =]

Reply:
That sounds good to me too. We should cover at least mailto: and tel:, I'd think. But that's another PR.